### PR TITLE
Split settings into "Basic" and "Advanced" sections

### PR DESCRIPTION
**Pull request by leon-richardt**
Pull request checklist:

- [ ] `CHANGELOG.md` was updated, if applicable

# Description
Progress towards https://github.com/Chatterino/chatterino2/issues/1956, refer to the linked issue for more information.

**TODO**:
- [x] Devise a way to hide "Advanced" section (see `ExpandableLayout` class)
- [x] Decide which settings should go into the "Basic" and "Advanced" section
- [ ] Consistent search behavior
- [ ] Decide on a strategy for "Advanced Settings" button UX
- [ ] Explore other strategies: Advanced Settting on different page, Advanced Settings in different tab
- [ ] Document which settings have been moved

### CODE REVIEW

**Comment by leon-richardt** [comment]
Points mentioned by @gempir in chat:
- Opening the "Advanced Settings" dropdown results in unintuitive UX as the new content is barely visible. Should *at the least* scroll down to the new settings.
- Increase spacing to "Miscellaneous"  section. (Looks too related.)
- Button should be more noticeable in general.
- Why not use an existing style for advanced settings? (Either an own settings page or an own tab on the "General" page.)
  - I tried to emulate Chromium's style of hiding advanced setting. I like the idea of these settings being "out of the way" for most users so I opted for this approach. Curios about further input though.

**Comment by pajlada** [comment]
Haven't looked at the PR yet, but another "advanced settings" software user is OBS, might be worth looking at that for some inspiration

File: src/widgets/helper/ExpandableLayout.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        this->toggleButton_->setArrowType(Qt::ArrowType::UpArrow);
+        this->content_->show();
+    }
+    else
+    {
+        // Collapse
+        this->toggleButton_->setArrowType(Qt::ArrowType::DownArrow);
+        this->content_->hide();
+    }
```
Comment (23rd) [inline src/widgets/helper/ExpandableLayout.cpp:51]:
```suggestion
    this->toggleButton_->setArrowType(expand ? Qt::ArrowType::UpArrow
                                             : Qt::ArrowType::DownArrow);
    this->content_->setVisible(expand);
```

Comment (leon-richardt) [reply src/widgets/helper/ExpandableLayout.cpp]: Thanks, I applied your suggestion in 0dd6170 via a rebase.

**Comment by leon-richardt** [comment]
Closing until I come up with a better idea to organize this

**Pull request closed without merging**